drv/sidecar-mainboard-controller/src/fan_modules.rs

```diff
@@ -0,0 +1,154 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{Addr, MainboardController, Reg};
+use bitfield::bitfield;
+use drv_fpga_api::{FpgaError, FpgaUserDesign, WriteOp};
+use userlib::FromPrimitive;
+use zerocopy::{AsBytes, FromBytes};
+
+bitfield! {
+    #[derive(Copy, Clone, PartialEq, Eq, FromPrimitive, AsBytes, FromBytes)]
+    #[repr(C)]
+    pub struct FanModuleStatus(u8);
+    pub enable, set_enable: 0;
+    pub led, set_led: 1;
+    pub present, _: 2;
+    pub power_good, _: 3;
+    pub power_fault, _: 4;
+    pub power_timed_out, _: 5;
+}
+
+/// Each fan module contains two individually controlled fans
+pub const NUM_FAN_MODULES: usize = 4;
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum FanModuleLedState {
+    Off = 0,
```

Comment:
Do we rely on these explicit values anywhere?  If they're used as part of FPGA communication, then there should be a note to that effect; otherwise, I'd suggest removing them.

Reply:
We do not! I don't know why my brain opted to do that in this case. Removed!